haskell/private/actions/compile.bzl

```diff
@@ -425,11 +425,38 @@ def compile_binary(
     coverage_data = []
     if inspect_coverage:
         c.args.add_all(_hpc_compiler_args(hs))
+
+        # Generate the list of instrumentation files (.mix) for coverage
+        # This list is sensible to file naming, which must match the internal
+        # module name because ghc will generate a ".mix" file named based on
+        # the module name.
+
+        # This is an haskell binary, so we *need* to find a "Main" module
+        main_found = False
+
         for src_file in srcs:
             module = module_name(hs, src_file)
+
+            # We introduces a few heuristics so users may not name their "Main" module "Main.hs".
```

Comment:
A similar heuristic was implemented [here](https://github.com/tweag/rules_haskell/pull/1281/files#diff-d764c440aa72bbd13b6c9de1e20ffe376bce5317a1d0f14c7c43a38ebf1be2caR121-R129), outlined [here](https://github.com/tweag/rules_haskell/pull/1281/files#diff-d764c440aa72bbd13b6c9de1e20ffe376bce5317a1d0f14c7c43a38ebf1be2caR83-R93). It covers an additional edge-case: If `main_function` is set and defines a different module name than `Main`. Could you re-use that here?